### PR TITLE
Feature/michael/add events

### PR DESCRIPTION
**Pull request by Michael-B-D**
for issue create new event #32

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -26,6 +26,7 @@ share/python-wheels/
 .installed.cfg
 *.egg
 MANIFEST
+calender-venv/
```
Comment (yammesicka) [inline .gitignore:29]: Please add a virtualenv section, and put this line under it :)

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (yammesicka) [inline .vscode/settings.json:1]: Add to `.gitignore`

File: app/database/database.py [hunk 3]
```diff
@@ -5,7 +5,7 @@
 from sqlalchemy.orm import sessionmaker
 
 
-SQLALCHEMY_DATABASE_URL = os.getenv("DATABASE_CONNECTION_STRING")
+SQLALCHEMY_DATABASE_URL = "sqlite:///./calendar_app.db"
```
Comment (yammesicka) [inline app/database/database.py:8]: Please import this line from config.py

File: app/database/models.py [hunk 4]
```diff
@@ -22,8 +22,23 @@ class Event(Base):
 
     id = Column(Integer, primary_key=True, index=True)
     title = Column(String)
+    start_date = Column(DateTime)
+    end_date = Column(DateTime)
+    VC_link = Column(String)
```
Comment (yammesicka) [inline app/database/models.py:27]: vc_link (prefer lowercase even on acronyms)

File: app/database/schemas.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
+from datetime import date
+from pydantic import BaseModel
+
+
+class Record(BaseModel):
```
Comment (yammesicka) [inline app/database/schemas.py:5]: This line would probably not run(?)

File: app/main.py [hunk 6]
```diff
@@ -1,18 +1,38 @@
...
 
+from app.database import models
+from app.database.models import Event
+from app.database.database import SessionLocal, engine
+
+models.Base.metadata.create_all(bind=engine)
+
+
+
+def get_db():
```
Comment (yammesicka) [inline app/main.py:18]: Try not to add content to main.py. Also, this line can be found on `dependencies.py`

File: app/main.py [hunk 7]
```diff
@@ -30,3 +50,64 @@ def profile(request: Request):
         "username": current_username,
         "events": upcouming_events
     })
+
+
+@app.get("/profile/{user_id}/EditEvent", response_class=HTMLResponse)
```
Comment (yammesicka) [inline app/main.py:55]: move to routers/profile.py

File: app/main.py [hunk 8]
```diff
@@ -30,3 +50,64 @@ def profile(request: Request):
...
+async def insert_info(request: Request):
+    """Get request and return an html File"""
+    return templates.TemplateResponse("editevent.html",{"request": request})
+
+
+@app.post("/profile/{user_id}/EditEvent") # this func is soupose to change with the PR of Ode and Efrat and it will be change
```
Comment (yammesicka) [inline app/main.py:61]: Prefer new line comments instead of inline comments

**Comment by yammesicka** [comment]
Please add calendar_app.db to your `.gitignore`

File: app/templates/base.html [hunk 9]
```diff
@@ -6,7 +6,6 @@
 
     <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.0-beta1/dist/css/bootstrap.min.css" rel="stylesheet"
         integrity="sha384-giJF6kkoqNQ00vy+HMDP7azOuL0xtbfIcaT9wjKHr8RbDVddVHyTfAAsrekwKmP1" crossorigin="anonymous">
```
Comment (yammesicka) [inline app/templates/base.html:8]: Why did you remove the `style.css` line?

File: app/routers/create_event.py [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+def create_item(user_id: int, event_title: str = Form(...), location: Optional[str] = Form(None), from_date: Optional[datetime] = Form(...),
+                to_date: Optional[datetime] = Form(...), link_vc: str = Form(None), content: str = Form(None), repeated_event: str = Form(None)):
+    # requierd args - title, from_date, to_date, user_id
+    # check validation for the return valuo 
+    # if the prosess success return json with True boolean arg and the event_id
+    # return False otherwith
+    prosess = False
+    event_valuo = {'title': event_title, "location": location, "from_date": from_date, "to_date": to_date, "link_vc":link_vc, "content": content, 
+                    "repeated_event": repeated_event}
+    # try:
```
Comment (yammesicka) [inline app/routers/create_event.py:39]: Please clean the code from comments.

File: app/routers/create_event.py [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+    if check_validation(event_title, from_date, to_date):
+        pass
+        # x = Event.content
+    # insert the valuo to the DB with SQLalcemy 
+
+#  except Exception as identifier:
+    # pass
+    return {from_date < to_date}
```
Comment (yammesicka) [inline app/routers/create_event.py:47]: Set of True or False?

File: app/routers/create_event.py [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+
+
+@app.get("/profile/EditEvent", response_class=HTMLResponse)
+async def insert_info(request: Request):
+    return templates.TemplateResponse("editevent.html",{"request": request})
+
+
+@app.post("/profile#id={user_id}/EditEvent")
+def create_item(user_id: int, event_title: str = Form(...), location: Optional[str] = Form(None), from_date: Optional[datetime] = Form(...),
+                to_date: Optional[datetime] = Form(...), link_vc: str = Form(None), content: str = Form(None), repeated_event: str = Form(None)):
```
Comment (yammesicka) [inline app/routers/create_event.py:31]: Please add return type. Also, keep in mind that having function with more than 3 arguments is usually a code smell. Can you please address it?

File: app/routers/create_event.py [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+app = FastAPI()
+
+# app.mount("/static", StaticFiles(directory="static"), name="static")
+
+
+templates = Jinja2Templates(directory="../templates")
+
+
+@app.get("/profile/EditEvent", response_class=HTMLResponse)
+async def insert_info(request: Request):
```
Comment (yammesicka) [inline app/routers/create_event.py:25]: Add a type annotation for the return type.

File: app/main.py [hunk 14]
```diff
@@ -30,3 +50,64 @@ def profile(request: Request):
...
+def create_event(user_id: int, event_title: str = Form(...), location: Optional[str] = Form(None), from_date: Optional[datetime] = Form(...),
```
Comment (yammesicka) [inline app/main.py:62]: See comment that follows about having functions with many parameters.

File: app/main.py [hunk 15]
```diff
@@ -30,3 +50,64 @@ def profile(request: Request):
...
+            error_msg = "Error, Your date is invalid"
+    except Exception as e:
+        error_msg = e
+    finally:
+        return {success, new_event, error_msg}
+
+
+def check_validation(start_time, end_time):
+    """Check if the start_date is smaller then the end_time"""
+    if start_time < end_time:
```
Comment (yammesicka) [inline app/main.py:89]: `return start_time < end_time`

File: app/main.py [hunk 16]
```diff
@@ -30,3 +50,64 @@ def profile(request: Request):
...
+                    VC_link = values['VC_link'],
+                    content = values['content'],
+                    location = values['location'],
+                    owner_id = values['owner_id']
+                    )
+        db.add(new_event)   
+        db.commit()
+        db.refresh(new_event)
+        return new_event
+    except Exception as e:
```
Comment (yammesicka) [inline app/main.py:111]: Please catch a specific exception.

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 15 above)
Comment (PureDreamer) [reply app/main.py:89]: In the docstring you wrote start_date , it should be strart_time.
Also I would suggest writing earlier and not smaller.

File: app/routers/create_event.py (hunk 12 above)
Comment (Michael-B-D) [reply app/routers/create_event.py:31]: Sorry, I don't understand..
if you mean that I receive json\kwargs that not possible because the setting on receive data from 'form' in fastapi see here - https://fastapi.tiangolo.com/tutorial/request-forms/
